DataProcessing/CryptoCoarseFundamentalUniverseDataDownloader.cs

```diff
@@ -0,0 +1,213 @@
+                    {
+                        _existingSecurities[dataSymbol].SetMarketPrice(new Tick { Value = (decimal)coarseByDate[dataSymbol][^2] });
+
+                        decimal? usdVol = null;
+
+                        // In case there might be missing data
+                        try
+                        {
+                            var volume = _dataByDate[date][dataSymbol][^1];
+                            usdVol = GetUSDVolume(volume, _market, _quoteCurrency[dataSymbol], _existingSecurities.Values.ToList());
+                        }
+                        catch
+                        {
+                            Log.Trace($"No USD-{dataSymbol.Value} rate conversion available on {date}.");
+                        }
+
+                        coarseByDate[dataSymbol].Add(Extensions.SmartRounding((decimal)usdVol));
+                    }
+
+                    // Save to file
+                    var fileContent = coarseByDate.Select(x => 
+                        {
+                            var sid = x.Key.ID;
+                            return $"{sid},{sid.Symbol},{string.Join(",", x.Value)}";
+                        })
+                        .ToList();
+                    var finalPath = Path.Combine(_destinationFolder, $"{date}.csv");
+                    var finalFileExists = File.Exists(finalPath);
+
+                    var lines = new HashSet<string>(fileContent);
+                    if (finalFileExists)
+                    {
+                        foreach (var line in File.ReadAllLines(finalPath))
+                        {
+                            lines.Add(line);
+                        }
+                    }
+
+                    var finalLines = lines.OrderBy(x => x.Split(',').First());
+                    File.WriteAllLines(finalPath, finalLines);
+                }
+            }
+            catch (Exception e)
+            {
+                Log.Error(e);
+                return false;
+            }
+
+            Log.Trace($"CryptoCoarseFundamentalUniverseDataConverter.ConvertToUniverseFile(): Finished Processing");
+            return true;
+        }
+
+        /// <summary>
+        /// Helper method to get USD volume from quote currency-intermittent pair
+        /// </summary>
+        /// <param name="baseVolume">The volume of symbol at date in quote currency</param>
+        /// <param name="market">The market exchange</param>
+        /// <param name="quoteCurrency">The quote currency of the Symbol</param>
+        /// <param name="existingSecurities">List of existing securities available to exchange rate</param>
+        /// <return>
+        /// Volume in USD
+        /// </return>
+        public static decimal? GetUSDVolume(
+            decimal? baseVolume,
+            string market,
+            string quoteCurrency,
+            List<Security> existingSecurities
+        )
+        {
+            var targetCurrency = market == Market.Binance ? "BUSD" : "USD";
```

Comment:
Why is this required, seems like a bug hiding somewhere? I'd expect to see target currency always be USD?

Reply:
Binance doesn't support USD as quote currency, and most of their pairs are associated with BUSD which is their stable coin to USD.

Reply:
> Binance doesn't support USD as quote currency

Ah right! it should though? I think once this fix is merged https://github.com/QuantConnect/Lean/pull/6259/files this shouldn't be required